crates/net/network/src/network.rs

```diff
@@ -221,7 +221,7 @@ impl NetworkInfo for NetworkHandle {
     }
 
     fn network_status(&self) -> NetworkStatus {
-        todo!()
+        NetworkStatus::default()
```

Comment:
we should be able to populate this already, most of this is available in the `SessionManager`

so I guess we need to make this async, same as the other async functions in the `NetworkHandle`

Reply:
Gotcha! Seems that we'll need to pass the `SessionManager` to the NetworkHandle, and in turn we'll probably have to make the `SessionManager` shareable (it does not implement `Copy`, seems like it should look like the `NetworkHandle itself`) right?

Reply:
Aight, gave it a shot on https://github.com/paradigmxyz/reth/pull/923/commits/ddf8dc375bf5cd597c0a2df5235f399133231deb , looks fine imo but introduces a `Result<T,E>` on fetching the network status which feels kinda out of place with all other rpc responses.